packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -46,7 +53,81 @@ export async function loadConfig(cwd: string) {
 
   const options = await architectHost.getOptionsForTarget(buildTarget);
   if (!options) throw new Error("Not able to find options for build target.");
-  return options;
+
+  // options has to be of type ApplicationBuilderOptions when the builder is an application builder
+  const applicationBuilderOptions: ApplicationBuilderOptions = Object.assign(
+    {} as ApplicationBuilderOptions,
+    options,
+  );
+
+  return applicationBuilderOptions;
+}
+
+// populate file/directory paths we need inside app hosting output directory
+export function populateOutputBundleOptions(config: ApplicationBuilderOptions): OutputPathOptions {
+  const outputPath = config.outputPath;
+  // normalized output path structure
+  const normalizedOutputPath = {
+    browser: "browser",
+    server: "server",
+    media: "media",
+    ...(typeof outputPath === "string" ? undefined : outputPath),
+    base: normalize(resolve(typeof outputPath === "string" ? outputPath : outputPath.base)),
+  };
+  const outputBundleDir = resolve(".apphosting");
+
+  return {
+    bundleYamlPath: resolve(outputBundleDir, "bundle.yaml"),
+    outputDirectory: outputBundleDir,
+    baseDirectory: resolve(normalizedOutputPath.base),
+    outputBaseDirectory: resolve(outputBundleDir, "dist"),
+    serverFilePath: resolve(outputBundleDir, "dist", normalizedOutputPath.server, "server.mjs"),
+    browserDirectory: resolve(outputBundleDir, "dist", normalizedOutputPath.browser),
+  };
+}
+
+// Run build command
+export const build = (cwd = process.cwd()) =>
+  new Promise<void>((resolve, reject) => {
+    // enable JSON build logs for application builder
+    process.env.NG_BUILD_LOGS_JSON = "1";
+    const childProcess = spawn("npm", ["run", "build"], { cwd, shell: true, stdio: "inherit" });
+    childProcess.on("exit", (code) => {
+      if (code === 0) return resolve();
+      reject();
+    });
+  });
+
+// move the base output directory, which contains the server and browser bundle directory, and prerendered routes
+// as well as generating bundle.yaml
```

Comment:
better to use multi-line comments instead for better support in the code editor for all functions

Reply:
done